src/main/java/bio/terra/cli/businessobject/resource/BqDataset.java

```diff
@@ -112,13 +112,19 @@ public static BqDataset createControlled(CreateBqDatasetParams createParams) {
   }
 
   /** Update a BigQuery dataset referenced resource in the workspace. */
-  public void updateReferenced(UpdateResourceParams updateParams) {
-    if (updateParams.name != null) {
-      validateEnvironmentVariableName(updateParams.name);
+  public void updateReferenced(UpdateReferencedBqDatasetParams updateParams) {
+    if (updateParams.getResourceParams().name != null) {
+      validateEnvironmentVariableName(updateParams.getResourceParams().name);
+    }
+    if (updateParams.getProjectId() != null) {
```

Comment:
I don't understand the purpose of this test. If `updateParams.getProjectId()` is null, you would be assigning null to the projectId. But isn't the projectId null by construction?

Said another way, why wouldn't you always: `this.projectId = updateParams.getProjectId()` regardless of the null state?

Reply:
because this is used for updatereference. if the user doesn't specify an update param, we don't want to override the existing param to null.